src/components/common/ColonyTotalFunds/ColonyTotalFunds.tsx

```diff
@@ -82,24 +66,30 @@ const ColonyTotalFunds = () => {
 
   const isSupportedColonyVersion = true;
 
-  // if (!data || !currentToken || isLoadingTokenBalances) {
-  //   return (
-  //     <MiniSpinnerLoader
-  //       className={styles.main}
-  //       loadingText={MSG.loadingData}
-  //       titleTextValues={{ hasCounter: false }}
-  //     />
-  //   );
-  // }
+  const totalTokenBalance = useMemo(() => {
+    if (balances?.items && currentToken) {
+      return (
+        balances.items
+          /*
+           * If the domain is not set, then we're dealing with "All Domains" (id 0)
+           */
+          .filter((balance) => balance?.domain === null)
+          .find(
+            (balance) =>
+              balance?.token?.tokenAddress === currentToken.token.tokenAddress,
+          ) || { balance: '0' }
+      );
+    }
+    return { balance: '0' };
+  }, [balances, currentToken]);
```

Comment:
Could all the logic from `useColonyContext` down to here become `useColonyTotalFunds`, like `useColonyFundsClaims`? 

Not a big point, especially if we'll be revisiting this all again shortly. More trying to get a sense of whether that's a pattern we like in general for components with more than a few lines of logic above the view (and thus something we can implement generally going forward).

Reply:
Sure, I'm happy to extract as much logic as I can into hooks